app/bundles/EmailBundle/Tests/Model/SendEmailToContactTest.php

```diff
@@ -725,4 +725,102 @@ function () use ($emailMock) {
+        $factoryMock = $this->getMockBuilder(MauticFactory::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $factoryMock->expects($this->at(3))->method('getParameter')->with('mailer_spool_type')->willReturn('memory');
+        $factoryMock->expects($this->at(1))->method('getParameter')->willReturn('');
+
+        $factoryMock->method('getLogger')
+            ->willReturn(
+                new NullLogger()
+            );
+
+        $mockEm = $this->createMock(EntityManager::class);
+
+        $factoryMock->method('getEntityManager')
+            ->willReturn($mockEm);
+
+        $mockDispatcher = $this->getMockBuilder(EventDispatcher::class)
+            ->getMock();
+        $mockDispatcher->method('dispatch')
+            ->willReturnCallback(
+                function ($eventName, EmailSendEvent $event) {
+                    $lead = $event->getLead();
+
+                    $tokens = [];
+                    foreach ($lead as $field => $value) {
+                        $tokens["{contactfield=$field}"] = $value;
+                    }
+                    $tokens['{hash}'] = $event->getIdHash();
+
+                    $event->addTokens($tokens);
+                }
+            );
+        $factoryMock->method('getDispatcher')
+            ->willReturn($mockDispatcher);
+        $routerMock = $this->getMockBuilder(Router::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $factoryMock->method('getRouter')
+            ->willReturn($routerMock);
+
+        $copyRepoMock = $this->getMockBuilder(CopyRepository::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $emailModelMock = $this->getMockBuilder(EmailModel::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $emailModelMock->method('getCopyRepository')
+            ->willReturn($copyRepoMock);
+
+        $factoryMock->method('getModel')
+            ->willReturn($emailModelMock);
+
+        $mailHelper = $this->getMockBuilder(MailHelper::class)
+            ->setConstructorArgs([$factoryMock, $mailer])
+            ->setMethods(null)
+            ->getMock();
+
+        // Enable queueing
+        $mailHelper->enableQueue();
+
+        $statRepository = $this->getMockBuilder(StatRepository::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $statRepository->method('saveEntity')
+            ->willReturnCallback(
+                function (Stat $stat) {
+                    $tokens = $stat->getTokens();
+                    $this->assertGreaterThan(1, count($tokens));
+                    $this->assertEquals($stat->getTrackingHash(), $tokens['{hash}']);
+                }
+            );
+
+        $dncModel = $this->getMockBuilder(DoNotContact::class)
+            ->disableOriginalConstructor()
+            ->getMock();
```

Comment:
These 3 rows can be simplified into `$factoryMock = $this->createMock(DoNotContact::class);`